src/rgw/rgw_admin.cc

```diff
@@ -8108,7 +8108,34 @@ int main(int argc, const char **argv)
       cerr << "ERROR: sync.init() returned ret=" << ret << std::endl;
       return -ret;
     }
-    ret = sync.read_sync_status(dpp());
+
+    rgw_bucket_sync_status full_status;
+    RGWBucketSyncPolicyHandlerRef handler;
+    const auto& sources = handler->get_sources();
+    const auto source_it = sources.find(source_zone);
+    if (source_it == sources.end()) {
+      cerr << "ERROR: source zone: '" << source_zone << "' could not be found" << std::endl;
+      return 0;
+    }
+    const auto& pipes = source_it->second;
+    if (pipes.pipe_map.empty()) {
+      cerr << "ERROR: no destination zones for source zone: '" << source_zone << "'" << std::endl;
+      return 0;
+    }
+    const auto& pipe = pipes.pipe_map.begin()->second;
```

Comment:
i still don't understand this new pipe stuff, but i feel like you're duplicating stuff in `RGWBucketPipeSyncStatusManager`/`RGWRemoteBucketManager`, and we may be better off changing `sync.read_sync_status()` so that it reads the full_status from the same source. `RGWRemoteBucketManager` has a `full_status` field (though unused?) that may help with that

that said, `read_sync_status()` itself is really confusing because it loops over `source_mgrs` but only returns one shard status map

Reply:
not needed anymore, just calling `sync.read_sync_status(dpp());`